xarray/backends/cfgrib_.py

```diff
@@ -50,28 +56,29 @@ class CfGribDataStore(AbstractDataStore):
     """
     Implements the ``xr.AbstractDataStore`` read-only API for a GRIB file.
     """
-    def __init__(self, ds, lock=False):
-        self.ds = ds
-        self.lock = lock
-
-    @classmethod
-    def from_path(cls, path, lock=False, **backend_kwargs):
+    def __init__(self, filename, lock=None, **backend_kwargs):
         import cfgrib
-        return cls(ds=cfgrib.open_file(path, **backend_kwargs), lock=lock)
+        if lock is None:
+            lock = ECCODES_LOCK
+        self.lock = ensure_lock(lock)
+        backend_kwargs['filter_by_keys'] = tuple(backend_kwargs.get('filter_by_keys', {}).items())
```

Comment:
This looks a little surprising to me -- can we simply pass on `backend_kwargs` directly to cfgrib?

Reply:
This is a ugly hack araund the fact that `filter_by_keys` is a dict but `CachingFileManager` accepts only hashable `backend_kwargs` because they are passed to `_HashedSequence`.

`filter_by_keys` has a very nice dict semantics, so I'd prefer not to change it.

Reply:
OK, that makes please. I think the need to hash arguments used to open files with CachingFileManager is unavoidable, so this is a reasonable workaround. But please add a comment explaining this.

Reply:
Ok, I added a comment and made the code more explicit.